src/contracts/SimpleAdjudicator.ts

```diff
@@ -0,0 +1,46 @@
+import detectNetwork from 'web3-detect-network';
+import contract from 'truffle-contract';
+import BN from 'bn.js';
+// @ts-ignore
+import simpleAdjudicatorArtifact from 'fmg-simple-adjudicator/contracts/SimpleAdjudicator.sol';
+
+import connectedWeb3 from '../wallet/web3';
+
+export async function deploySimpleAdjudicator(channelId: string, amount: BN) {
+  const truffleContract = await setupContract();
+
+  let contractBytecode;
+  contractBytecode = truffleContract.bytecode;
+
+  Object.keys(truffleContract.links).forEach(linkName => {
+    const replace = `__${linkName}_________________________________`;
+    contractBytecode = contractBytecode.replace(new RegExp(replace,"g"), truffleContract.links[linkName].substr(2));
+  });
+
+  const web3Contract = new connectedWeb3.eth.Contract(simpleAdjudicatorArtifact.abi, connectedWeb3.eth.defaultAccount);
+
+  const tx = web3Contract.deploy({
+    data: contractBytecode,
+    arguments: [channelId],
+  });
+
+  const deployedContract = await tx.send({ from: connectedWeb3.eth.defaultAccount, value: amount.toString() });
```

Comment:
@andrewgordstewart Elsewhere in the app, we use the `yield` + the saga framework to block on asynchronous calls. Do you see us using the `await` pattern in other parts of the app going forward?

Reply:
I think this would improve testability, along with @tomclose's work on moving more logic into reducers.

Since reducers are synchronous, you couldn't move this logic into a reducer, for instance.

However, by moving this logic out of a saga, we could more easily write unit tests for, eg., `deployContract`.

So, I think we should explore this idea: extracting some async logic out of sagas, and using the (more common) `await` pattern.